modules/system/console/asset/AssetCompile.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+namespace System\Console\Asset;
+
+use Symfony\Component\Process\Process;
+use System\Classes\CompilableAssets;
+use System\Classes\PackageJson;
+use Winter\Storm\Console\Command;
+use Winter\Storm\Support\Facades\File;
+use Winter\Storm\Support\Str;
+
+abstract class AssetCompile extends Command
+{
+    /**
+     * @var string The console command description.
+     */
+    protected $description = 'Mix and compile assets';
+
+    /**
+     * PackageJson object holding the contents of the active package.json
+     */
+    protected PackageJson $packageJson;
+
+    /**
+     * Name of config file i.e. mix.webpack.js, vite.config.js
+     */
+    protected string $configFile;
+
+    /**
+     * File path being watched, used for cleanup by mix:watch
+     */
+    protected string $watchingFilePath;
+
+    public function compileHandle(string $type): int
+    {
+        // Exit early if node_modules isn't available yet
+        if (!File::exists(base_path('node_modules'))) {
+            $this->error('The Node dependencies are not available, try running mix:install first.');
```

Comment:
Should this still be `mix:install` as the recommended command?

Reply:
No it should not, see: https://github.com/wintercms/winter/pull/1141/commits/6409346e5528ff539d5e3137a1b2057bc9f1f47e